File: apps/api/src/services/get-user-id.ts
import { db } from '@/lib/db'
import { supabase } from '@/lib/supabase'
import { Request } from 'express'

export const getUserId = async (
  req: Request,
) => {
  const token = req.headers.authorization

  if (!token) {
    return null
  }

  try {
    const payload = await supabase.auth.getUser(token)

    if (!payload.data.user) {
      return null
    }

    const user = await db
      .selectFrom('users')
      .select(['id'])
      .where('provider_id', '=', payload.data.user.id)
      .executeTakeFirst()

    if (!user) {
      return null
    }

    return user.id
  } catch (error) {
    return null
  }
}

File: apps/backend/backend/services/get_user_id.py
from backend.models.user import User
from backend.libs.supabase import supabase
from backend.libs.db import get_db
from flask import Request

def get_user_id(request: Request):
    token = request.headers.get('Authorization')
    if not token:
        return None
    
    try:
        payload = supabase.auth.get_user(token)
        if not payload.user:
            return None
        
        db = next(get_db())
        user = db.query(User).where(User.provider_id == payload.user.id).first()
        return user.id if user else None
    except:
        return None